dune-project

```diff
@@ -37,7 +37,7 @@
   (gen_js_api
    (= 1.1.1))
   (base
-   (>= v0.14.1))
+   (and (= 0.15.1)))
```

Comment:
Shouldn't it be something like `>= 0.15 && < 0.16`?

Reply:
I tried, but get this for Windows:
```
  [ERROR] No solution for vscode-ocaml-platform & vscode:   * Missing dependency:
              - base < 0.16
              no matching version
```

I'm still not sure what causes an actual problem. I think it is because base 0.16 was released to opam-repository, but it's not in opam-repository-mingw. I was trying to experiment with different version constraints to see if I can make it work somehow.

When I constrained to `= 15.1` I got this for all platforms:
```
  # Detecting depexts using vars: arch=x86_64, os=win32, os-distribution=cygwinports, os-family=windows
  [ERROR] No solution for vscode-ocaml-platform & vscode: The following dependencies couldn't be met:
            - vscode-ocaml-platform -> base < v0.9.0
                no matching version
```
which weird, because I don't know where 0.9.0 comes from...

Reply:
I tried, but get this for Windows:
```
  [ERROR] No solution for vscode-ocaml-platform & vscode:   * Missing dependency:
              - base < 0.16
              no matching version
```

I'm still not sure what causes an actual problem. I think it is because base 0.16 was released to opam-repository, but it's not in opam-repository-mingw. I was trying to experiment with different version constraints to see if I can make it work somehow.

When I constrained to `= 15.1` I got this for all platforms:
```
  # Detecting depexts using vars: arch=x86_64, os=win32, os-distribution=cygwinports, os-family=windows
  [ERROR] No solution for vscode-ocaml-platform & vscode: The following dependencies couldn't be met:
            - vscode-ocaml-platform -> base < v0.9.0
                no matching version
```
which is weird and I don't know where 0.9.0 comes from...

Reply:
If it doesn't work on Windows, that's ok. Keep in mind that just be can't build the plugin on Windows, doesn't mean we can't run it there (we just publish js artifacts). It's enough to build on MacOS and Linux if we're bottlenecked by upstream somehow.

Reply:
ok, let's leave like this then

Reply:
I had to make that change for the ocaml-lsp CI to have the windows runner access recent Merlin releases:

https://github.com/ocaml/ocaml-lsp/commit/806767c545b19f9d2fae6223563c5cb7532aa231

It might be related to your issue.

Reply:
Yup, I made similar changes in #1101. Currently, this step looks very similar to yours:
https://github.com/ocamllabs/vscode-ocaml-platform/blob/c41d801fe9cbf2ea3c14fa934bb71f1e3c9aa5dc/.github/workflows/main.yml#L35

I think the problem is that the base library is 0.16 in default and 0.15.1 in mingw repo. I tried couple of hacks, but it didn't work out. Maybe you have some ideas?

Reply:
I think that this release broke the build:
https://github.com/ocaml/opam-repository/tree/master/packages/base/base.v0.16.0

Reply:
The only way I was able to make it work is to fork the opam-repository, revert it to the state before base v0.16 release, and use this fork as a default repo for windows step.
This idea was taken from this comment: https://github.com/ocaml/opam/issues/4003#issuecomment-539567990

I'm not sure if there are simpler ways to achieve the goal. I tried pins/locks/constraints, but for some reason it didn't work out. Maybe we can use this fork repo approach? WDYT?

Reply:
I think it is not worthy to go to such an extent to get the CI to pass since there is no necessity to have this project building on Windows as @rgrinberg said. Maybe we could just deactivate it ?

Also, did you considered actually upgrading to base 0.16 ?

Reply:
The problem with base 0.16 is that it fails to build on Windows. This is a reason why I made this PR. base was constrained to >= 0.15, so we built it with the 0.16 after it's release, which passed on mac and linux but failed on windows. The error message was pretty lengthy:
```
#=== ERROR while compiling base.v0.16.0 =======================================#
# context     2.0.10 | win32/x86_64 | ocaml-variants.4.14.1+mingw64c | git+https://github.com/ocaml/opam-repository.git
# path        D:/a/vscode-ocaml-platform/vscode-ocaml-platform/_opam/.opam-switch/build/base.v0.16.0
# command     D:\a\vscode-ocaml-platform\vscode-ocaml-platform\_opam\bin\dune.exe build -p base -j 2
# exit-code   1
# env-file    D:/.opam/log/base-3496-ae662f.env
# output-file D:/.opam/log/base-3496-ae662f.out
### output ###
# File "src/dune", line 10, characters 0-500:
# 10 | (library (name base) (public_name base)
# 11 |  (ocamlopt_flags :standard (:include ocamlopt-flags))
# 12 |  (libraries base_internalhash_types caml sexplib0 shadow_stdlib)
# 13 |  (c_flags :standard -D_LARGEFILE64_SOURCE (:include mpopcnt.sexp))
# 14 |  (c_names bytes_stubs exn_stubs int_math_stubs hash_stubs am_testing)
# 15 |  (preprocess no_preprocessing)
# 16 |  (lint
# 17 |   (pps ppx_base ppx_base_lint -check-doc-comments -type-conv-keep-w32=both
# 18 |    -apply=js_style,base_lint,type_conv,cold))
# 19 |  (js_of_ocaml (javascript_files runtime.js)))
# (cd _build/default && D:\a\vscode-ocaml-platform\vscode-ocaml-platform\_opam\bin\ocamlopt.opt.exe -w -40 -g -shared -linkall -I src -o src/base.cmxs src/base.cmxa)
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_b45414_dune\dyndllde3800.o::(.data+0x3378): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_b45414_dune\dyndll11e42a.o::(.data+0x2e60): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_b45414_dune\dyndlld26bdc.o::(.data+0xe60): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_b45414_dune\dyndll250902.o::(.data+0xc0): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_b45414_dune\dyndll2c9fa9.o::(.data+0x1b90): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_b45414_dune\dyndll2fd195.o::(.data+0x1b50): more undefined references to `caml_csel_value' follow
# collect2: error: ld returned 1 exit status
# ** Fatal error: Error during linking
# 
# File "caml_startup", line 1:
# Error: Error during linking (exit code 2)



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build base v0.16.0
+- 
```
so I thought it's easier to stay with 0.15.1 and wait until it's patched in mingw repo.

Reply:
Deactivating the Windows job might be a reasonable option. Let's see what's Rudi's opinion about it.

Reply:
Deactivating or leaving it as failing is fine with me